src/main/java/com/faforever/client/domain/GameBean.java

```diff
@@ -12,47 +12,69 @@
+@Value
 public class GameBean {
   public static final Integer OBSERVERS_TEAM = -1;
   public static final Integer NO_TEAM = 1;
 
-  private final StringProperty host = new SimpleStringProperty();
+  StringProperty host = new SimpleStringProperty();
   @ToString.Include
-  private final StringProperty title = new SimpleStringProperty();
-  private final StringProperty mapFolderName = new SimpleStringProperty();
-  private final StringProperty featuredMod = new SimpleStringProperty();
+  StringProperty title = new SimpleStringProperty();
+  StringProperty mapFolderName = new SimpleStringProperty();
+  StringProperty featuredMod = new SimpleStringProperty();
   @EqualsAndHashCode.Include
   @ToString.Include
-  private final IntegerProperty id = new SimpleIntegerProperty();
-  private final IntegerProperty maxPlayers = new SimpleIntegerProperty();
-  private final StringProperty leaderboard = new SimpleStringProperty();
-  private final ObjectProperty<Integer> ratingMin = new SimpleObjectProperty<>(null);
-  private final ObjectProperty<Integer> ratingMax = new SimpleObjectProperty<>(null);
-  private final BooleanProperty passwordProtected = new SimpleBooleanProperty();
-  private final StringProperty password = new SimpleStringProperty();
+  IntegerProperty id = new SimpleIntegerProperty();
+  IntegerProperty maxPlayers = new SimpleIntegerProperty();
+  StringProperty leaderboard = new SimpleStringProperty();
+  ObjectProperty<Integer> ratingMin = new SimpleObjectProperty<>();
+  ObjectProperty<Integer> ratingMax = new SimpleObjectProperty<>();
+  BooleanProperty passwordProtected = new SimpleBooleanProperty();
+  StringProperty password = new SimpleStringProperty();
   @ToString.Include
-  private final ObjectProperty<GameStatus> status = new SimpleObjectProperty<>();
-  private final ObjectProperty<VictoryCondition> victoryCondition = new SimpleObjectProperty<>();
-  private final ObjectProperty<OffsetDateTime> startTime = new SimpleObjectProperty<>();
-  private final BooleanProperty enforceRating = new SimpleBooleanProperty(false);
-  private final ObjectProperty<GameType> gameType = new SimpleObjectProperty<>();
+  ObjectProperty<GameStatus> status = new SimpleObjectProperty<>();
+  ObjectProperty<VictoryCondition> victoryCondition = new SimpleObjectProperty<>();
+  ObjectProperty<OffsetDateTime> startTime = new SimpleObjectProperty<>();
+  BooleanProperty enforceRating = new SimpleBooleanProperty();
+  ObjectProperty<GameType> gameType = new SimpleObjectProperty<>();
   /**
    * Maps a sim mod's UID to its name.
    */
-  private final ObjectProperty<Map<String, String>> simMods = new SimpleObjectProperty<>(Map.of());
-  private final ObjectProperty<Map<Integer, List<PlayerBean>>> teams = new SimpleObjectProperty<>(Map.of());
+  ObjectProperty<Map<String, String>> simMods = new SimpleObjectProperty<>(Map.of());
+  ObjectProperty<Map<Integer, Set<PlayerBean>>> teams = new SimpleObjectProperty<>(Map.of());
```

Comment:
Yes because for these properties we do not want to make changes directly to the maps. So they are designed to be immutable.

Instead we are observing when the map itself changes. Which is why this is an observable value wrapping a map.

Although later I changed this to a Map property which is the same thing with a few extra methods. But it is still wrapping an unmodifiable map.